### PR TITLE
Firebase analytics - Login

### PR DESCRIPTION
**Pull request by Sashen943**
- Analytics Manager added into presenter
- Functions in Presenter added to track analytics
- Injected into Login Navigation 
- Implemented in Login View Controller
- Tests added for new functionality in presenter
- General code clean up of login feature and tests:
>removed trailing white spaces
> added MARK comments
> refactored flow and renamed variables

### CODE REVIEW

File: Showcase-iOS/Login/View/LoginNavigationController.swift [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
-        loginViewController?.loginPresenter = loginPresenter
-        loginViewController?.sharedApplication = dependencyContainer.resolve(SharedApplicationDelegate.self)
+        if let loginPresenter = dependencyContainer.resolve(LoginPresentable.self) as? LoginPresenter {
+            loginPresenter.loginViewer = loginViewController
+            loginPresenter.wireframe = dependencyContainer.resolve(WireframeDelegate.self)
+            loginViewController?.loginPresenter = loginPresenter
+            if let analyticsManager = dependencyContainer.resolve(AnalyticsManager.self) as? AnalyticManagerImplementation {
+                 loginPresenter.analyticsManager = analyticsManager
+            }
+            loginViewController?.sharedApplication = dependencyContainer.resolve(SharedApplicationDelegate.self)
```
Comment (pjwelcome) [inline Showcase-iOS/Login/View/LoginNavigationController.swift]: This should not be inside this if let statement as it does not rely on login presenter

**Review by pjwelcome — CHANGES_REQUESTED** [review]

File: Showcase-iOS/Login/View/LoginNavigationController.swift (hunk 1 above)
Comment (Sashen943) [reply Showcase-iOS/Login/View/LoginNavigationController.swift]: done :)

File: Showcase-iOS/Login/View/LoginViewController.swift [hunk 2]
```diff
@@ -39,130 +45,160 @@ class LoginViewController: UIViewController, MFMailComposeViewControllerDelegate
...
     func showPasswordValidationFailure(withError error: AuthenticationError) {
-        self.showAlert(withTitle:"Invalid password.", "Please provide a strong password.")
+        self.showAlert(withTitle:"Invalid password.",
+                       "Please provide a strong password.")
     }
-    
+
     func showInvalidInputsFailure(withError error: AuthenticationError) {
-        self.showAlert(withTitle:  "Invalid inputs." , "Please make sure that you've provided valid inputs for username and password.")
+        self.showAlert(withTitle:  "Invalid inputs." ,
```
[the remark this answers is not in the document]
Comment (Sashen943) [reply Showcase-iOS/Login/View/LoginViewController.swift]: done

File: Showcase-iOS/Login/View/LoginNavigationController.swift [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-        var loginPresenter = dependencyContainer.resolve(LoginPresentable.self)
-        loginPresenter?.loginViewer = loginViewController
-        loginPresenter?.wireframe = dependencyContainer.resolve(WireframeDelegate.self)
-        loginViewController?.loginPresenter = loginPresenter
-        loginViewController?.sharedApplication = dependencyContainer.resolve(SharedApplicationDelegate.self)
+        if let loginPresenter = dependencyContainer.resolve(LoginPresentable.self) as? LoginPresenter {
+            loginPresenter.loginViewer = loginViewController
+            loginPresenter.wireframe = dependencyContainer.resolve(WireframeDelegate.self)
+            loginViewController?.loginPresenter = loginPresenter
+            if let analyticsManager = dependencyContainer.resolve(AnalyticsManager.self) as? AnalyticManagerImplementation {
```
[the remark this answers is not in the document]
Comment (Sashen943) [reply Showcase-iOS/Login/View/LoginNavigationController.swift]: done

**Review by pjwelcome — APPROVED** [review]

**Pull request merged**